specs/Badge.md

```diff
@@ -0,0 +1,106 @@
+# Badge
+
+## Background
+
+A badge is an additional visual discriptor for UI elements.It can be used to denote numerical value, status or general information.
+
+## Prior Art
+
+_Include background research done for this component_
+
+- [OpenUI Research](https://open-ui.org/components/badge.research)
+- [AntD](https://ant.design/components/badge/)
+- [Atlassian](https://atlassian.design/components/badge/examples)
+- [Bootstrap](https://getbootstrap.com/docs/4.3/components/badge/)
+- [Fast](https://explore.fast.design/components/fast-badge)
+- [MaterialUI](https://material-ui.com/components/badges/#badge)
+- [Lightning](https://www.lightningdesignsystem.com/components/badges/)
+- [React Native](https://reactnativeelements.com/docs/badge/)
+
+## Sample Code
+
+```jsx
+  <Badge>
+    <Text>My Custom Badge</Text>
+  </Badge>
+  <Badge
+    status="success"
+    style={{ position: 'absolute', top: -4, right: -4 }}
+  />
+```
+
+## Variants
+
+`Badge` can have several variations.
+
+- Appearance: `default`, `rounded` and `circular`
+- Size: `smallest`, `smaller`, `small`, `medium`, `large`, `larger`.
+- Styles: `filled`, `outline`, `ghost`, `tint`, `inverted filled`
+
+## API
+
+| Property Name | Type      | Default Value | Description                  |
+| ------------- | --------- | ------------- | ---------------------------- |
+| `size`        | `enum`    | `medium`      | The Badge size               |
+| `status`      | `enum`    | `none`        | The Badge status color       |
```

Comment:
This `status` enum seems specific only to certain types of badges. The basic badge should probably just allow a color value to be set instead of having an enum built in.

Then the variants like `PresenceBadge` can define an enum like this and apply appropriate colors based on the enum.

Reply:
We had a discussion about color prop and that is not going in. User has the shred colors listed in the spec has reference but will not be supported directly by the Badge, the only badge colors that will be implemented out of the box will be the ones used for Status or Presence ( semantic colors ) and any other color should be the user to create some custom badge

Reply:
Yeah having a color prop has problems of its own. But having a `status` enum on the base badge has problems too, because the child badges would have two ways to set their color--what does it mean to set both `status` and `presence` on a `PresenceBadge`?

I think having the base badge have no opinion about colors is best. It'd just need some way for the child types like `PresenceBadge` to pick colors of their own.

If we want to support status colors like this, perhaps we need a `StatusBadge`?

Reply:
Yeah, the basic badge will have no opinion about color, I am ok with `StatusBadge` although I don't see a problem with having `status` and `presence` to set `color` and `icon`